libraries/aws/shadow/include/types/aws_iot_shadow_types.h

```diff
@@ -381,7 +366,7 @@ typedef enum AwsIotShadowCallbackType
  */
 typedef struct AwsIotShadowCallbackParam
 {
-    AwsIotShadowCallbackType_t callbackType; /**< @brief Reason for invoking the Shadow callback function to provide context. */
+    uint32_t callbackType; /**< @brief Reason for invoking the Shadow callback function to provide context. */
```

Comment:
I would prefer `enum` to `uint32_t` here.  Is the underlying problem that this library was using the old `enum` in arithmetic?  If so, the *best* solution would be make that not be the case so that we could still use an `enum`; however if we don't have time to do that then I suppose this could be acceptable.

Reply:
Yes, also for consistency with other libraries (which use enum).

Reply:
My comment wasn't clear. Casting other essential types into enums is forbidden in MISRA 10.5. 
The original solution was this
https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/743/files 

I believe the concern was the lookup was possibly too expensive.